lib/elixir/lib/list.ex

```diff
@@ -636,6 +636,28 @@ defmodule List do
     end
   end
 
+  @doc """
+  Returns a list containing only the elements that `list1` and `list2` have in
+  common.
+
+  ## Examples
+
+      iex> List.intersection([1, 2, 3, 4, 5], [3, 4, 5, 6, 7])
+      [3, 4, 5]
+
+      iex> List.intersection([4, 2, 5, 3, 1], [12, 1, 9, 5, 0])
+      [5, 1]
+
+      iex> List.intersection([1, 2, 3, 4, 5], [6, 7, 8, 9, 0])
+      []
+
+  """
+  @spec intersection(list(), list()) :: list()
+  def intersection(list1, list2) do
```

Comment:
```suggestion
  def intersection(list1, list2) do
    list1 -- list1 -- list2
  end
```

Reply:
I agree. I used the `temp` variable for the sake of readability.

Reply:
> I agree. I used the `temp` variable for the sake of readability.

I guess parentheses could be used,  to save the creating of a variable